File: SortingAlgorithms/InsertSort.java
public class InsertSort {

	public InsertSort() {}

	public int[] sort(int[] arr) {
		for (int i = 1; i < arr.length; i++) {
			int key = arr[i];
			int j = i - 1;

			while (j >= 0 && arr[j] > key) {
				arr[j + 1] = arr[j];
				j--;
			}
			
			arr[j + 1] = key;
		}
		return arr;
	}

	public static void main(String[] args) {

		InsertSort insertSort = new InsertSort();
		int[] arr = {2, 54, 2, 1, 43, 3};

		for (int ele: arr) {
			System.out.print(ele + " ");
		}
		
		arr = insertSort.sort(arr);
		System.out.println();
		
		for (int ele: arr) {
			System.out.print(ele + " ");
		}
	}
}

File: SortingAlgorithmsImpl/InsertSort.cpp
#include <iostream>
#include <vector>
#include "/Algos/HelperClass/Vecs.h"

using namespace std;

class InsertSort {

	public:

		void sort(vector<int>& vec) {

		    for (size_t i = 1; i < vec.size(); i++) {
		        int key = vec[i];
		        size_t j = i;
		
		        while (j > 0 && vec[j - 1] > key) {
		            vec[j] = vec[j - 1];
		            j--;
		        }
		
		        vec[j] = key;
		    }
		}
};

int main() {

	InsertSort insertSort;
	Vecs vecs;
		
	vector<int> vec = {3, 32, 1, 34, 767};

	vecs.printVector(vec);
	cout << endl;

	insertSort.sort(vec);

	vecs.printVector(vec);	
	cout << endl;
	
}
